### ISSUE TITLE
Allow setting of EC2 "Name" tag

### ISSUE THREAD

**Issue by mslavin-tibco** [aws/karpenter#944]
**Tell us about your request**
Allow the setting of the EC2 "Name" tag via spec.provider.tags. Currently, the tag.go -> ManagedTagsFor function always hard codes this as `Name: karpenter.sh/<cluster-name>`. 

**Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?**
If you have multiple provisioners, say for provisioning large or specialty EC2 types, it's nice to be able to identify these differences in the EC2 console by the name of the EC2. If you attempt to include the Name tag in the provisioner spec, the webhook says the tag key "Name" is reserved. 

**Are you currently working around this issue?**
No

**Additional context**
Release: v0.5.1

**Attachments**
None

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by ellistarn** [comment]
This is a great request. I think it should be straight forward to enable this. We probably should make a better default. If we included provisioner name, do you still want control over the name?

Thoughts on something like `karpenter.sh/clusters/$clustername/provisioners/$provisionername`. Would you be willing to open a PR?

**Comment by mslavin-tibco** [comment]
Hi @ellistarn, IMO, the current implementation is probably fine for the default value. If we need to discover resources created by Karpenter and that requires a known tag, it seems reasonable to add something like "app.kubernetes.io/created-by: karpenter.sh/xyz...". But, forcing the Name tag to have a fixed structure seems restrictive. Given that the "Name" tag is used for display purposes in the AWS console, organizations (or SRE) may have requirements as to what that should be. Thoughts on using some reasonable default if the following is not supplied in the provisioner spec?
```
spec:
  provider: 
    tags:
      Name: "xyz"
``` 
In other words, honor the supplied name tag if provided and supply some default if it's not. Re: opening a PR, I can take a crack at the implementation, but I was having some issues trying to build the go code locally.

**Comment by ellistarn** [comment]
This seems like a solid default to me. As long as we're comfortable with the fact that tags are applied to all created resources (e.g. instances, LTs).

Ping me on kubernetes slack and I can help you with any build issues.

**Comment by agrrh** [comment]
Just to clarify, templating works for default Name only, right?

Setting some custom value for Name in my YAML, I must hard-code cluster and provisioner names.

**Comment by mslavin-tibco** [comment]
@agrrh See PR ^^^ for additional details, since you are in control with how the provisioner CR is submitted, if it's wrapped in some helm chart, it can be set via
```
  provider: 
    tags:
      Name: {{ .Values.myCustomProvisioner }}
```
If the "Name" tag is not provided in your provisioner CR, then the default Name tag will be set using
```
  "Name": "karpenter.sh/cluster/%s/provisioner/%s"
```
where the cluster name and provisioner are dynamically populated.

**Comment by agrrh** [comment]
Looks like things became broken in 0.5.5. Following config results in different node naming:

```
spec:
  provider:
    tags:
      Name: my-custom-name
```

- 0.5.4 produces "my-custom-name", as expected
- 0.5.5 results in "karpenter.sh/provisioner-name/example-provisioner"

**Comment by mslavin-tibco** [comment]
@ellistarn I'm guessing it was this change that broke things https://github.com/aws/karpenter/commit/a304cb51f5adf6c4b45d77c6a615d2ea62477937#diff-78dcc34ff9df6a5c682f80310e41a5ba740d8c1f6a03736929f9d6b1cb16d0fa?

**Comment by ellistarn** [comment]
https://github.com/aws/karpenter/pull/1211

**Comment by v-yarotsky** [comment]
FWIW, if the Name tag is specified via a custom Launch Template, Karpenter will still set it to `karpenter.sh/provisioner-name/<name>` on the instance.

**Comment by cx-IliyaY** [comment]
Q here
Do i need to have `AWSNodeTemplate` for each `Provisioner` to use this
```provider: 
        tags:
          Name: "xyz"
```
because when i'm using in my provisioner `providerRef` i can't use also `provider`....
and i can use `provider` when i don't have any `AWSNodeTemplate`

So how can i actually Tag my EC2 name?
